src/lib/utils/synapseTypes/EntityBundle.ts

```diff
@@ -45,3 +51,64 @@ export type EntityBundleRequest = {
   includeThreadCount?: boolean
   includeRestrictionInformation?: boolean
 }
+
+/**
+ * Maps the fields of an Entity bundle to the EntityBundleRequest field that is required to return that field
+ */
+interface BundleFieldToRequestFieldMapping
+  extends Record<keyof _EntityBundle, keyof EntityBundleRequest> {
+  entity: 'includeEntity'
+  entityType: 'includeEntity'
+  annotations: 'includeAnnotations'
+  permissions: 'includePermissions'
+  path: 'includeEntityPath'
+  hasChildren: 'includeHasChildren'
+  accessControlList: 'includeAccessControlList'
+  fileHandles: 'includeFileHandles'
+  tableBundle: 'includeTableBundle'
+  rootWikiId: 'includeRootWikiId'
+  benefactorAcl: 'includeBenefactorACL'
+  doiAssociation: 'includeDOIAssociation'
+  fileName: 'includeFileName'
+  threadCount: 'includeThreadCount'
+  restrictionInformation: 'includeRestrictionInformation'
+}
+
+/**
+ * Given the following types:
+ * @param TRequestConst: a constant request object with boolean values (e.g. an EntityBundleRequest)
+ * @param TRequestField: a key of the request object
+ * @param TResultObject: a result object (e.g. EntityBundle)
+ * @param TResultField: a key of the result object
+ *
+ * If the constant object has the specified field set to true, then the resulting type will be TResultObject[TResultField].
+ *
+ * Otherwise, the resulting type is never, indicating that the field is not present in the result.
+ */
+type IncludeOnlyIfRequested<
+  TRequestConst extends Record<string, boolean>,
+  TRequestField extends keyof TRequestConst,
+  TResultObject,
+  TResultField extends keyof TResultObject,
+> = TRequestConst[TRequestField] extends true
+  ? TResultObject[TResultField]
+  : never
```

Comment:
This might cause other problems with dynamic request objects or not passing a generic (neither of which we are doing today), so I'm wondering if this should be `TResultObject[TResultField] | undefined`, at the cost of losing the strictness of `never`

Reply:
Could also consider

```
TRequestConst[TRequestField] extends true
  ? TResultObject[TResultField] :
 	(TResultObject[TResultField] extends boolean ? 
	 TResultObject[TResultField] | undefined
	  : never)
```

Reply:
Difficult to read with the nested ternary operators, but logic is sound

Reply:
Unfortunately TypeScript doesn't give us a ton of tools to make types more readable 😵‍💫